api/filesys/fsnode.go

```diff
@@ -396,8 +427,12 @@ func (n *fsNode) Write(p []byte) (c int, err error) {
 		return 0, fmt.Errorf(
 			"cannot write content to non-file '%s'", n.Path())
 	}
-	n.content = make([]byte, len(p))
-	return copy(n.content, p), nil
+	if n.offset == nil {
+		return 0, fmt.Errorf("cannot write to closed file '%s'", n.Path())
+	}
+	n.content = append(n.content[:*n.offset], p...)
+	*n.offset = len(n.content)
+	return len(p), nil
```

Comment:
@monopole any opinion on what file opening mode we should emulate here? Before, `Write` replaced the entire content every time (independent of previous reads or writes). Now, if I understand correctly:
- if the file has been entirely read, Write will append
- if the file hasn't been read, Write will overwrite, potentially partially, from the beginning
- if the file has been partially read (or rewritten), Write will overwrite, potentially partially, from the current position

While that may be strictly closer to a particular set of real file mode flags, it's also probably less easy to understand.

Reply:
I added better docs on `Open` including an example of the expected behavior. Let me know if that helps.

Reply:
i understand, thanks for improving fsNode.